src/open_inwoner/utils/url.py

```diff
@@ -58,3 +61,23 @@ def prepend_next_url_param(url: str, next_url: str) -> str:
         next_url = prepend_next_url_param(next_url, current_next)
     u.args.set("next", next_url)
     return u.url
+
+
+def uuid_from_url(
+    url: str, allow_multiple: bool = False
+) -> uuid.UUID | list[uuid.UUID] | None:
+    """Extract UUID(s) from a given URL"""
+
+    pattern = r"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+    matches = re.findall(pattern, url, re.IGNORECASE)
```

Comment:
Given that this is likely to be called in loops, we could pre-compile:

```suggestion

_UUID_PATTERN = re.compile(r"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}")

def uuid_from_url(
    url: str, allow_multiple: bool = False
) -> uuid.UUID | list[uuid.UUID] | None:
    """Extract UUID(s) from a given URL"""

    matches = _UUID_PATTERN.findall(pattern, url, re.IGNORECASE)
```